app/controllers/DataSetController.scala

```diff
@@ -176,7 +178,7 @@ class DataSetController @Inject()(val messagesApi: MessagesApi) extends Controll
       case (server, name, token, team) =>
         for {
           _ <- DataSetService.checkIfNewDataSetName(name) ?~> Messages("dataSet.name.alreadyTaken")
-          _ <- ensureTeamAdministration(request.identity, team)
+          _ <- Fox.assertBoolean(request.identity.isTeamManagerOrAdminOf(ObjectId.fromBsonId(team))) ?~> Messages("team.admin.notAllowed")
```

Comment:
Why not stay with the new `ensureTeamAdministration`? Seems to be equivalent.

Reply:
Yes, will do.
(the explanation is that I first wanted to remove the `ensureTeamAdministration` entirely because I found it a little superfluos and the name a little confusing, but then I saw that it was used in so many places that I stopped removing it. Also, I guess does help reduce clutter a little bit)